src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java

```diff
@@ -0,0 +1,171 @@
+        var file = Files.createFile(filePath).toFile();
+        // write content to the created file
+        FileUtils.write(file, currentLocalFileContent);
+
+        var testRepoUrl = new GitUtilService.MockFileRepositoryUrl(testRepo.originRepoFile);
+        exercise.setTestRepositoryUrl(testRepoUrl.toString());
+        doReturn(gitService.getRepositoryByLocalPath(testRepo.localRepoFile.toPath())).when(gitService).getOrCheckoutRepository(testRepoUrl.getURL(), true);
+    }
+
+    @AfterEach
+    public void tearDown() throws IOException {
+        database.resetDatabase();
+        reset(gitService);
+        LocalRepository.resetLocalRepo(testRepo);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldGetFiles() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        var files = request.getMap(testRepoBaseUrl + exercise.getId() + "/files", HttpStatus.OK, String.class, FileType.class);
+        assertThat(files).isNotEmpty();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldGetFile() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        LinkedMultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("file", currentLocalFileName);
+        var file = request.get(testRepoBaseUrl + exercise.getId() + "/file", HttpStatus.OK, byte[].class, params);
+        assertThat(file).isNotEmpty();
+        assertThat(new String(file)).isEqualTo(currentLocalFileContent);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldCreateFile() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        LinkedMultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/newFile"))).isFalse();
+        params.add("file", "newFile");
+        request.postWithoutResponseBody(testRepoBaseUrl + exercise.getId() + "/file", HttpStatus.OK, params);
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/newFile"))).isTrue();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldCreateFolder() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        LinkedMultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/newFolder"))).isFalse();
+        params.add("folder", "newFolder");
+        request.postWithoutResponseBody(testRepoBaseUrl + exercise.getId() + "/folder", HttpStatus.OK, params);
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/newFolder"))).isTrue();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldRenameFile() throws Exception {
```

Comment:
Maybe also test renaming a folder.

Reply:
i added a test case for renaming a folder